DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCapture.cs

```diff
@@ -0,0 +1,275 @@
+                    m_depthBufferBits == other.m_depthBufferBits &&
+                    m_copyDepth == other.m_copyDepth &&
+                    m_depthCopyFormat == other.m_depthCopyFormat &&
+                    m_depthCopyMode == other.m_depthCopyMode;
+            }
+            
+            public static bool operator ==(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => lhs.Equals(rhs);
+
+            public static bool operator !=(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => !(lhs == rhs);
+        }
+
+        /// <summary>
+        /// Called once textures have been captured and are ready to be consumed.
+        /// </summary>
+        public event Action<ScriptableRenderContext, CameraCapture> onTexturesReady = delegate {};
+        
+        /// <summary>
+        /// <para>
+        /// The captured color texture. Its format is defined by <see cref="description"/>.
+        /// </para>
+        ///
+        /// <para>
+        /// Note: while the texture may be configured for MSAA, Unity will always use an implicitly resolved version
+        /// of the texture for APIs such <see cref="CommandBuffer.CopyTexture(RenderTargetIdentifier, RenderTargetIdentifier)"/>
+        /// and see <see cref="Texture.GetNativeTexturePtr"/>.
+        /// </para>
+        ///
+        /// <remarks>
+        /// To avoid unnecessary texture copies, this refers directly to <see cref="Camera.targetTexture"/>.
+        /// </remarks>
+        /// </summary>
+        public RenderTexture cameraTexture => m_cameraTexture;
+        
+        /// <summary>
+        /// The captured depth texture. Its format is defined by <see cref="description"/>.
+        /// </summary>
+        public RenderTexture depthTexture => m_depthTexture;
+
+        /// <summary>
+        /// Defines the color and depth textures. The textures are automatically disposed and created on change.
+        /// </summary>
+        public CameraCaptureDescription description
+        {
+            get => m_description;
+            set
+            {
+                if (m_description != value)
+                {
+                    m_description = value;
+                    Refresh();
+                }
+            }
+        }
+
+        [SerializeField]
+        CameraCaptureDescription m_description = CameraCaptureDescription.Default;
+
+        DepthCopy m_depthCopy;
+        Camera m_camera;
+        RenderTexture m_cameraTexture;
+        RenderTexture m_depthTexture;
+        
+#if DEBUG
+        bool m_HasSetSecondNames;        
+#endif
+
+#if UNITY_EDITOR
+        CameraCaptureDescription m_lastDescription = CameraCaptureDescription.Default;
+        bool m_RefreshFlag; // Ensures thread safety (OnValidate is on a different thread)
+        
+        void OnValidate()
+        {
+            if (m_lastDescription != m_description)
+            {
+                m_lastDescription = m_description;
```

Comment:
I wasn't aware that `OnValidate()` runs on a different thread, but if it does, then you have another problem. It means that you have a data race on `m_description` if `Refresh()` happens to be executing at the same time, since assigning a struct is not an atomic operation. You need a `lock` around the usages of `m_description`.

Reply:
Keeping a flag for simplicity by having it trigger a check instead of a refresh directly